### PR TITLE
Add support for SAM E70 revision B

### PR DESCRIPTION
**Pull request by aurel32**
This PR imports the HAL for the SAM E70 revision B, and adds support for it. This is a per-requisite for fixing #12711.

### CODE REVIEW

**Comment by aurel32** [comment]
@dgloeck FYI, that should solve part of PR #12711.

**Comment by dgloeck** [comment]
@aurel32, your revision B headers still need the changes from #13204.
Or do you want to wait until Microchip makes a new release of the headers?

**Comment by aurel32** [comment]
> @aurel32, your revision B headers still need the changes from #13204.
> Or do you want to wait until Microchip makes a new release of the headers?

@dgloeck, I was waiting for PR #13204 to be merged. Now that it has been done, I have just pushed an updated version.

File: soc/arm/atmel_sam/same70/soc.c
Comment (mnkp) [inline soc/arm/atmel_sam/same70/soc.c]: Maybe it would be better to do a full check as in
```
	if (CHIPID->CHIPID_CIDR != CHIP_CIDR) {
```

It makes the code simpler and also covers cases where we try to flash firmware on a device with different flash or RAM size.

Comment (aurel32) [reply soc/arm/atmel_sam/same70/soc.c]: Thanks for the idea, I just implemented that.

**Review by mnkp — APPROVED** [review]

**Comment by dgloeck** [comment]
Is this Codeowners error a real problem blocking merging?

**Comment by mnkp** [comment]
Missing entry in CODEOWNERS file looks like a real issue. But this PR hasn't been merged so far as it has no 1.14 milestone, neither 'bug' nor 'TSC' label. I couldn't find any open issue but as far as I remember this PR is required to solve a bug related to gmac driver on sam_e70_xplained board. If that's a case we should try to merge it for 1.14.

**Comment by aurel32** [comment]
The CODEOWNER check is relatively new, and the files for the SAM E70 rev A do not have a codeowner assigned either. If it blocks the PR I can add myself as a codeowner, but I feel it should rather done independently of the revision A or B.

**Comment by mnkp** [comment]
@aurel32 you are working actively and have a lot of experience. I believe you're the best candidate to take ownership of any code related to Atmel SoCs.

**Comment by mnkp** [comment]
@aurel32 so what about this PR? Would be good to merge it.

**Comment by aurel32** [comment]
> @aurel32 so what about this PR? Would be good to merge it.

I have just pushed a new version with the same commit rebased onto master and an additional one adding myself as codeowner for the SAM E70 HAL.

Note that in the meantime Microchip has released a new pack, but it's probably better to update the HAL in a different PR.

**Comment by benpicco** [comment]
Looks like Shippable is not happy about me trying to resolve the mere conflict with master through that button on Github - just drop my merge commit and do a rebase instead :sweat_smile:

**Comment by aurel32** [comment]
> Looks like Shippable is not happy about me trying to resolve the mere conflict with master through that button on Github - just drop my merge commit and do a rebase instead sweat_smile

Done.

**Comment by benpicco** [comment]
Are the patches on top of the HAL (605b1b439ecf1cc57acb9050d82b100d4aa23b09) still needed with the new release of the HAL?

**Comment by mnkp** [comment]
I believe this PR should be back-ported to 1.14 release. #12711 which fixes a problem with the GMAC driver on SAM E70 Xplained Board using SAME70Q21B SoC relies on it. #12711 is a bugfix.

I'm however not clear about the rules related to backporting. Is anyone allowed to mark PR with the backport label. What are the exact criteria?

**Comment by aurel32** [comment]
> Are the patches on top of the HAL ([605b1b4](https://github.com/zephyrproject-rtos/zephyr/commit/605b1b439ecf1cc57acb9050d82b100d4aa23b09)) still needed with the new release of the HAL?

Not all of them, but some yes. The GMAC register offset issue is fixed only for rev B not for rev A. The missing TC defines is also fixed for rev B only, but using different names, so I am not sure how to deal with that.

In any case, I really think we should get this PR merged first and then work on the new version. Otherwise it will make things even more complicated to merge.

**Comment by aurel32** [comment]
> I believe this PR should be back-ported to 1.14 release. #12711 which fixes a problem with the GMAC driver on SAM E70 Xplained Board using SAME70Q21B SoC relies on it. #12711 is a bugfix.
> 
> I'm however not clear about the rules related to backporting. Is anyone allowed to mark PR with the backport label. What are the exact criteria?

I think one of the criteria is to get the current PR accepted in master first.

**Comment by nashif** [comment]
verified basic kernel tests pass on the e70 I have.

**Review by nashif — APPROVED** [review]

**Comment by mnkp** [comment]
So what about backporting this PR to 1.14 release?

**Pull request merged**
